deploy/openshift-clusters/roles/kcli/kcli-install/tasks/validate.yml

```diff
@@ -0,0 +1,121 @@
+---
+# Configuration validation for kcli-install
+
+- name: Validate cluster name
+  fail:
+    msg: "test_cluster_name must be defined and not empty"
+  when: test_cluster_name is not defined or test_cluster_name == ""
+
+- name: Validate topology
+  fail:
+    msg: "topology must be either 'fencing' or 'arbiter'"
+  when: topology is not defined or topology not in ['fencing', 'arbiter']
+
+- name: Validate domain
+  fail:
+    msg: "domain must be defined and not empty"
+  when: domain is not defined or domain == ""
+
+- name: Validate two-node configuration
+  fail:
+    msg: "For two-node OpenShift with fencing, ctlplanes must be 2 and workers must be 0"
+  when: ctlplanes != 2 or workers != 0
+
+- name: Validate memory requirements
+  fail:
+    msg: "Minimum memory requirement for OpenShift control plane nodes is 16GB (16384 MB)"
+  when: vm_memory < 16384
+
+- name: Validate disk size requirements
+  fail:
+    msg: "Minimum disk size requirement for OpenShift nodes is 120GB"
+  when: vm_disk_size < 120
+
+- name: Validate OpenShift version format
+  fail:
+    msg: "ocp_tag must be in format 'X.YY' (e.g., '4.16')"
+  when: ocp_tag is not match('^[0-9]+\.[0-9]+$')
+
+- name: Validate BMC credentials for fencing
+  fail:
+    msg: "BMC user and password must be defined for fencing configuration"
+  when: bmc_user == "" or bmc_password == ""
+
+- name: Validate CI configuration for CI builds
+  block:
+    - name: Check CI token when using CI builds
+      fail:
+        msg: "ci_token must be provided for CI builds when openshift_ci is false"
+      when: 
+        - ocp_version == "ci"
+        - not openshift_ci
+        - (ci_token is not defined or ci_token == "")
+
+    - name: Warn about OPENSHIFT_CI side effects
+      debug:
+        msg: "WARNING: openshift_ci=true avoids using CI_TOKEN but has side effects. See dev-scripts documentation."
```

Comment:
Does this still apply when we're deploying outside of dev-scripts?
Seems like both OPENSHIFT_CI and CI_TOKEN would be irrelevant here unless you were trying to supply a CI_TOKEN to pass to kcli because you want to use an CI payload.